notify/impl.go

```diff
@@ -270,16 +266,24 @@ func (n *Email) Notify(ctx context.Context, as ...*types.Alert) error {
 	}
 	defer c.Quit()
 
+	// We need to know the hostname for both auth and TLS.
+	host, _, err := net.SplitHostPort(n.conf.Smarthost)
+	if err != nil {
+		return fmt.Errorf("invalid address: %s", err)
+	}
+
+	if ok, _ := c.Extension("STARTTLS"); ok {
```

Comment:
Doesn't this open a downgrade attack vector? An attacker could just strip the STARTTLS field from the extension list and trick alertmanager into sending messages unencrypted.

What do you think about using a flag instead, so that alertmanager would send emails either always with TLS (and failing if it can't) or always without?